src/languages/sqf.js

```diff
@@ -1,2529 +1,2630 @@
+Description: Scripting language for the Arma game series
+Website: https://community.bistudio.com/wiki/SQF_syntax
+Category: scripting
+Last update: 07.01.2023, Arma 3 v2.11
+*/
+
+/*
+////////////////////////////////////////////////////////////////////////////////////////////
+	* Author: Leopard20
+	
+	* Description:
+	This script can be used to dump all commands to the clipboard.
+	Make sure you're using the Diag EXE to dump all of the commands.
+	
+	* How to use:
+	Simply replace the _KEYWORDS and _LITERALS arrays with the one from this sqf.js file.
+	Execute the script from the debug console.
+	All commands will be copied to the clipboard.
+////////////////////////////////////////////////////////////////////////////////////////////
+_KEYWORDS = ['if'];                                                //Array of all KEYWORDS
+_LITERALS = ['west'];                                              //Array of all LITERALS
+_allCommands = createHashMap;
+{
+	_type = _x select [0,1];
+	if (_type != "t") then {
+		_command_lowercase = ((_x select [2]) splitString " ")#(((["n", "u", "b"] find _type) - 1) max 0);
+		_command_uppercase = supportInfo ("i:" + _command_lowercase) # 0 # 2;
+		_allCommands set [_command_lowercase, _command_uppercase];
+	};
+} forEach supportInfo "";
+_allCommands = _allCommands toArray false;
+_allCommands sort true;                                            //sort by lowercase
+_allCommands = ((_allCommands apply {_x#1}) -_KEYWORDS)-_LITERALS; //remove KEYWORDS and LITERALS
+copyToClipboard (str (_allCommands select {_x regexMatch "\w+"}) regexReplace ["""", "'"] regexReplace [",", ",\n"]);
+*/
+
+export default function(hljs) {
+  // In SQF, a local variable starts with _
+  const VARIABLE = {
+    className: 'variable',
+    begin: /\b_+[a-zA-Z]\w*/
+  };
+
+  // In SQF, a function should fit myTag_fnc_myFunction pattern
+  // https://community.bistudio.com/wiki/Functions_Library_(Arma_3)#Adding_a_Function
+  const FUNCTION = {
+    className: 'title',
+    begin: /[a-zA-Z][a-zA-Z_0-9]*_fnc_[a-zA-Z_0-9]+/
+  };
+
+  // In SQF strings, quotes matching the start are escaped by adding a consecutive.
+  // Example of single escaped quotes: " "" " and  ' '' '.
+  const STRINGS = {
+    className: 'string',
+    variants: [
+      {
+        begin: '"',
+        end: '"',
+        contains: [ {
+          begin: '""',
+          relevance: 0
+        } ]
+      },
+      {
+        begin: '\'',
+        end: '\'',
+        contains: [ {
+          begin: '\'\'',
+          relevance: 0
+        } ]
+      }
+    ]
+  };
+
+  const KEYWORDS = [
+    'break', 'breakWith', 'breakOut', 'breakTo',
+    'case', 'catch', 'continue', 'continueWith',
+    'default', 'do',
+    'else', 'exit', 'exitWith',
+    'for', 'forEach', 'from',
+    'if',
+    'local',
+    'private',
+    'switch', 'step',
+    'then', 'throw', 'to', 'try',
+    'waitUntil', 'while', 'with'
+  ];
+
+  const LITERAL = [

```

Comment:
This is a LOT of literals.  **Consider** `variable.constant` and `variable.language` perhaps... `pi` (if it's math) seems like a reference to a literal (a constant), not a literal itself.

I won't force any changes here since I dunno the language, but just making you aware we have more scopes to differentiate these sorts of things.

Reply:
> This is a LOT of literals. **Consider** `variable.constant` and `variable.language` perhaps... `pi` (if it's math) seems like a reference to a literal (a constant), not a literal itself.
> 
> I won't force any changes here since I dunno the language, but just making you aware we have more scopes to differentiate these sorts of things.

Technically they're all commands (so they should be part of built-in array)
Again this is from previous contributions so I didn't want to change them.

Could you point me to the list of scopes you guys have?

Reply:
https://highlightjs.readthedocs.io/en/latest/css-classes-reference.html

> Technically they're all commands (so they should be part of built-in array).

Let's not do that. :-). For first party grammars we take a "wider perspective" view of such things.  Literal concepts that are non-reducible like true, false, null, nil, undefined... these are considered literals.  We prefer consistency in this across our grammars - rather than adhering to any particular grammars technicalities. 

Since `pi` is simply a label for 3.1459 etc it would be better thought of as a `variable.constant`... and the fact that what provides that is a `function pi()` doesn't really change that "wider perspective" categorization.